prov/sockets/src/sock_poll.c

```diff
@@ -142,8 +142,10 @@ static int sock_poll_poll(struct fid_poll *pollset, void **context, int count)
 						cntr_fid);
 			sock_cntr_progress(cntr);
 			pthread_mutex_lock(&cntr->mut);
-			if (atomic_get(&cntr->value) >=
-				atomic_get(&cntr->threshold)) {
+			if (atomic_get(&cntr->value) !=
+			    atomic_get(&cntr->last_read_val)) {
+				atomic_set(&cntr->last_read_val,
```

Comment:
Nope. This condition checks for the success case of fi_poll(). ie, current-cntr-val != last-read-val.
And in case of success, we always set the last-read-val as current value.